prow/plugins/lgtm/lgtm.go

```diff
@@ -253,45 +258,73 @@ func handle(wantLGTM bool, config *plugins.Configuration, ownersClient repoowner
 	hasLGTM := false
 	labels, err := gc.GetIssueLabels(org, repoName, number)
 	if err != nil {
-		log.WithError(err).Errorf("Failed to get the labels on %s/%s#%d.", org, repoName, number)
+		log.WithError(err).Errorf("Failed to get issue labels.")
 	}
 
 	hasLGTM = github.HasLabel(LgtmLabel, labels)
 
 	if hasLGTM && !wantLGTM {
 		log.Info("Removing LGTM label.")
-		return gc.RemoveLabel(org, repoName, number, LgtmLabel)
+		if err := gc.RemoveLabel(org, repoName, number, LgtmLabel); err != nil {
+			return err
+		}
+		log.Info("Removing tree-hash comment.")
+		botname, err := gc.BotName()
+		if err != nil {
+			return err
+		}
+		comments, err := gc.ListIssueComments(org, repoName, number)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to get issue comments.")
+		}
+		for _, comment := range comments {
+			m := addLGTMLabelNotificationRe.FindStringSubmatch(comment.Body)
+			if comment.User.Login == botname && m != nil {
+				if err := gc.DeleteComment(org, repoName, comment.ID); err != nil {
+					log.WithError(err).Errorf("Failed to delete comment.")
+				}
+			}
+		}
 	} else if !hasLGTM && wantLGTM {
 		log.Info("Adding LGTM label.")
 		if err := gc.AddLabel(org, repoName, number, LgtmLabel); err != nil {
 			return err
 		}
+		pr, err := gc.GetPullRequest(org, repoName, number)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to get pull request.")
+		}
+		SHA := *pr.MergeSHA
```

Comment:
Check that `MergeSHA` is non-nil before reading its value. This field is lazily calculated by GitHub and may not be populated.